src/main/scala/uk/gov/hmrc/SbtSassCompiler.scala

```diff
@@ -58,54 +58,88 @@ object SbtSassCompiler extends AutoPlugin {
+          Using(SassCompilerFactory.bundled()) { sassCompiler =>
+            logger.info(s"sbt-sass-compiler: Sass compiling for ${sassFilesFound.length} files")
+            val startInstant                                                   = System.currentTimeMillis
+            val eitherCompiledCssFiles: Seq[Either[Throwable, (String, Path)]] = sassFilesFound.map { sassFile =>
+              sassCompiler.setLoadPaths(sassLoadPaths) // no need to set a path of the current file as well
+              Try(sassCompiler.compileFile(sassFile).getCss) match {
+                case Success(compiledCss)      =>
+                  val cssFile = targetPath
+                    .resolve(sourcePath.relativize(sassFile.toPath))
+                    .resolveSibling(sassFile.base + ".css")
+                  Right((compiledCss, cssFile))
+                case Failure(compilationError) => Left(compilationError)
+              }
             }
-          }
 
-          val errors: Seq[Throwable]           = eitherCompiledCssFiles.flatMap(_.left.toOption)
-          val compiledCss: Seq[(String, Path)] = eitherCompiledCssFiles.flatMap(_.right.toOption)
-
-          if (errors.nonEmpty) {
-            throw new Exception(
-              s"${errors.length} error(s) whilst compiling Sass files\n${errors.mkString("\n")}"
-            )
-          } else {
-            val cssFiles: Seq[File] = compiledCss map { case (css, cssFile) =>
-              IO.createDirectory(cssFile.getParent.toFile)
-              IO.write(cssFile.toFile, css)
-              cssFile.toFile
+            val errors: Seq[Throwable]           = eitherCompiledCssFiles.flatMap(_.left.toOption)
+            val compiledCss: Seq[(String, Path)] = eitherCompiledCssFiles.flatMap(_.right.toOption)
+
+            if (errors.nonEmpty) {
+              throw new Exception(
+                s"sbt-sass-compiler: ${errors.length} error(s) whilst compiling Sass files\n${errors.mkString("\n")}"
+              )
+            } else {
+              val cssFiles: Seq[File] = compiledCss map { case (css, cssFile) =>
+                IO.createDirectory(cssFile.getParent.toFile)
+                IO.write(cssFile.toFile, css)
+                cssFile.toFile
+              }
+              val endInstant          = System.currentTimeMillis
+
+              logger.info(
+                s"sbt-sass-compiler: Sass compilation done in ${endInstant - startInstant} ms. Number of CSS files generated: ${cssFiles.length}"
+              )
+              cssFiles
             }
-            val endInstant          = System.currentTimeMillis
+          }.get
+        }
+
+        def getExistingCssFromTargetDirectory(): Seq[File] = {
+          Try {
+            Files
+              .walk(targetPath)
+              .iterator()
+              .asScala
+              .toSeq
+              .map(_.toFile)
+              .filter(_.getName.endsWith("css"))
+          } getOrElse Nil
+        }
```

Comment:
really small one, what about using the sbt file utils
```suggestion
        def getExistingCssFromTargetDirectory(): Seq[File] =
          targetPath.globRecursive("*.css").get
```
OR
```
val expectedOutputCSS = sassFilesFound.map(sassFile => 
  targetPath
    .resolve(sourcePath.relativize(sassFile.toPath))
    .resolveSibling(sassFile.base + ".css"))
```
then in the 
```
if (anySassFileChanged || (sassFilesFound.nonEmpty && existingCssFromTargetDirectory.isEmpty))
```
could be
```
def anyNotFound(files: Seq[File]): Boolean = files.exists(!_.exists)

if (anySassFileChanged || anyNotFound(expectedOutputCSS))
```

Reply:
might not be worth it though or more optimum 🤷‍♂️

Reply:
@oscarduignan I've added in your suggestion about using `globRecursive` to find compiled Sass files in `target`, good suggestion. For readability, I would admit my preference is the current:
`(sassFilesFound.nonEmpty && existingCssFromTargetDirectory.isEmpty)`

I find it's very understandable what it's doing then, but I'm aware that I'm not always the next judge for understanding code I've written!

Reply:
yup, happy to leave it like it is 🙌